esmvalcore/experimental/recipe_info.py

```diff
@@ -0,0 +1,271 @@
+        """
+        filename = Path(REFERENCES_PATH, f'{tag}.bibtex')
+        return cls(filename)
+
+    def __repr__(self):
+        """Return canonical string representation."""
+        return f'{self.__class__.__name__}({self._key!r})'
+
+    def __str__(self):
+        """Return string representation."""
+        return self.render(renderer='plaintext')
+
+    def _repr_markdown_(self):
+        """Represent using markdown renderer in a notebook environment."""
+        return self.render(renderer='markdown')
+
+    def render(self, renderer: str = 'plaintext') -> str:
+        """Render the reference.
+
+        Parameters
+        ----------
+        renderer : str
+            Choose the renderer for the string representation.
+            Must be one of: 'plaintext', 'markdown', 'html', 'latex'
+
+        Returns
+        -------
+        str
+            Rendered reference
+        """
+        style = 'plain'  # alpha, plain, unsrt, unsrtalpha
+        backend = pybtex.plugin.find_plugin('pybtex.backends', renderer)()
+        style = pybtex.plugin.find_plugin('pybtex.style.formatting', style)()
+
+        try:
+            formatter = style.format_entry(self._key, self._entry)
+            rendered = formatter.text.render(backend)
+        except Exception as err:
+            raise RenderError(
+                f'Could not render {self._key!r}: {err}') from None
+
+        return rendered
+
+
+class RecipeInfo():
+    """Contains Recipe metadata.
+
+    Parameters
+    ----------
+    path : pathlike
+        Path to the recipe.
+    """
+
+    def __init__(self, path: str):
+        self.path = Path(path)
+        if not self.path.exists():
+            raise FileNotFoundError(f'Cannot find recipe: `{path}`.')
+
+        self._data = None
+        self._authors = None
+        self._maintainers = None
+        self._projects = None
+        self._references = None
+        self._description = None
+
+    def __repr__(self) -> str:
+        """Return canonical string representation."""
+        return f'{self.__class__.__name__}({self.name!r})'
+
+    def _repr_markdown_(self) -> str:
+        """Represent using markdown renderer in a notebook environment."""
+        return self.render('markdown')
+
+    def __str__(self) -> str:
+        """Return string representation."""
+        return self.render('plaintext')
+
+    def to_markdown(self) -> str:
+        """Return markdown formatted string."""
+        return self.render('markdown')
+
+    def render(self, renderer: str = 'plaintext') -> str:
+        """Return formatted string.
+
+        Parameters
+        ----------
+        renderer : str
+            Choose the renderer for the string representation.
+            Must be one of 'plaintext', 'markdown'
+        """
+        bullet = '\n - '
+        string = f'## {self.name}'
+
+        string += '\n\n'
+        string += f'{self.description}'
+
+        string += '\n\n### Contributors'
```

Comment:
I think this should be authors, and mainainters is perhaps missing.

Reply:
Yep, good point!